libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/Actions/Microsoft.CancelAllDialogs.schema

```diff
@@ -12,19 +12,22 @@
         },
         "disabled": {
             "$role": "expression",
+            "type": [ "boolean", "string" ],
             "title": "Disabled",
             "description": "Optional condition which if true will disable this action.",
             "examples": [
                 "user.age > 3"
             ]
         },
         "eventName": {
+            "$role": "expression",
+            "type": "string",
             "title": "Event name",
-            "description": "Name of the event to emit.",
-            "type": "string"
+            "description": "Name of the event to emit."
         },
         "eventValue": {
-            "type": "object",
+            "$role": "expression",
+            "type": [ "object", "array", "number", "integer", "boolean", "string" ],
```

Comment:
>"type": [ "object", "array", "number", "integer", "boolean", "string" ], [](start = 12, length = 72)

Did you want to allow null for all of these expressions?

Reply:
no, I don't think so.  What's the scenario where you need a null?  if you don't set it, then it's not set.

---
In reply to: [367277858](https://github.com/microsoft/botbuilder-dotnet/pull/3253#discussion_r367277858) [](ancestors = 367277858)